app/services/sso/openid_connect.rb

```diff
@@ -25,6 +25,16 @@ def logout_url
       controller.extlogout_users_path
     end
 
+    def support_expiration?
+      true
+    end
+
+    def expiration_url
+      redirect_uri = "#{controller.main_app.extlogin_users_path}/redirect_uri"
+      "#{redirect_uri}?logout=#{CGI.escape(request.base_url + controller.main_app.extlogin_users_path)}"
+    end
+    alias_method :login_url, :expiration_url
+
```

Comment:
is this related to the ui notifications causing a redirect loop?

Reply:
Yeah, I can move this to another PR.

Reply:
looking at the session_expiry code again i now understand that we need to implement these methods for the redirect to go to the correct url, but these are indeed two separate issues- one for the notifications redirection to `/user/login` and the other for oidc to redirect on expiration to the correct url.